File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let map = new Map();
    for (let i = 0; i < nums.length; i++) {
        let match = target - nums[i];
        if(map.has(match)) {
            return [i, map.get(match)];
        }
        map.set(nums[i], i);
    }
};

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hash = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (hash.containsKey(diff)) {
                return new int[] { i, hash.get(diff) };
            }
            hash.put(nums[i], i);
        }
        return new int[] {};
    }
}